network/test/middleware_test.go

```diff
@@ -310,44 +326,72 @@ func (m *MiddlewareTestSuite) TestUnicastRateLimit_Messages() {
 			unittest.NetworkCodec().Encode,
 			network.ProtocolTypeUnicast)
 		require.NoError(m.T(), err)
-		err = m.mws[0].SendDirect(msg)
 
+		err = m.mws[0].SendDirect(msg)
 		require.NoError(m.T(), err)
 	}
 
 	// wait for all rate limits before shutting down middleware
-	unittest.RequireCloseBefore(m.T(), ch, 100*time.Millisecond, "could not stop on rate limit test ch on time")
+	unittest.RequireCloseBefore(m.T(), ch, 100*time.Millisecond, "could not stop rate limit test ch on time")
+
+	// sleep for 1 seconds to allow connection pruner to prune connections
+	time.Sleep(1 * time.Second)
+	// ensure connection to rate limited peer is pruned
+	isConnected, err = libP2PNodes[0].IsConnected(expectedPID)
+	require.NoError(m.T(), err)
+	require.False(m.T(), isConnected)
+
```

Comment:
Please apply the followings to both `TestUnicastRateLimit_Bandwidth` and `TestUnicastRateLimit_Messages`. 

1. Please use the following approach and check both the rate-limiting peer, as well as the rate-limited peer, do not have any connections to each other. We have to check _both sides_. [code taken from here](https://github.com/onflow/flow-go/blob/f20b8c85872bd07097fefde1fadb07bd5ef3ca07/network/internal/p2pfixtures/fixtures.go#L227-L228)
    ```go
        require.Equal(t, other.Host().Network().Connectedness(thisId), network.NotConnected)
        require.Empty(t, other.Host().Network().ConnsToPeer(thisId))
    ```
    Currently, the code takes the following approach and only checks connection on the side of the rate-limited node. But we need it on both sides. 
    ```go
  	  // ensure connection to rate limited peer is pruned
	  isConnected, err = libP2PNodes[0].IsConnected(expectedPID)
	  require.NoError(m.T(), err)
	  require.False(m.T(), isConnected)
    ```
    Also, please note that for sake of testing `Connecteness` is not enough as it does not factor in the secured but not upgraded connections, so we need to also add an extra check for upgraded ones: `require.Empty(t, other.Host().Network().ConnsToPeer(thisId))`
  
  2. Please try dialing the rate-limiting peer by the rate-limited peer right after rate-limiting is done, and check that no connection is established. To do so, please use the new approach on checking the connectivity (than reading the error). You may directly use the `ensureCommunicationSilenceAmongGroups` as written [here](https://github.com/onflow/flow-go/blob/cf027cd67e9185b17249dd93a48ed749d395e47e/network/p2p/connection/connection_gater_test.go#L366) noting that both nodes should be allowed to subscribe to the `channels.PushBlocks`.